code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -359,24 +361,56 @@
 		P.fire()
 		return P
 
-/mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
-	return iswallturf(T) || ismineralturf(T)
 
-/mob/living/simple_animal/hostile/proc/DestroySurroundings()
+/mob/living/simple_animal/hostile/proc/DestroyStructuresCycle(turf/T)
+	var/dir_to_target = get_dir(targets_from, get_cardinal_step_towards(targets_from, target))
+	var/list/dir_list = list(dir_to_target, turn(dir_to_target, 90), turn(dir_to_target, 270), turn(dir_to_target, 180))
+	cycle_loop:
+		for(var/dir in dir_list)
+			T = get_step(targets_from, dir)
+			for(var/a in T)
+				var/atom/A = a
+				if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
+					A.attack_animal(src)
+					break cycle_loop
+
+/mob/living/simple_animal/hostile/proc/DestroyPathToTarget()
+	if(environment_smash)
+		EscapeConfinement()
+		var/turf/T = get_cardinal_step_towards(targets_from, target)
+		if(T.Adjacent(targets_from))
```

Comment:
this if chain is... not good, and you don't actually need it

Reply:
i agree that escape confinement isnt really necessary but i thought it might be used in some edge case
same probably goes for environment smash
and i didnt think that Adjacent was necessary since the proc only gets 1 step away, but i figured id leave it in there
if its all not needed, ill get rid of it

Reply:
EscapeConfinement() is needed, but the if chain to check wall types is really bad and should be handled by the walls themselves.
with my above code, this proc should be something like this;
````
	if(environment_smash)
		EscapeConfinement()
		DestroyStructuresCycle()
````